src/wallet/wallet.h

```diff
@@ -986,4 +984,37 @@ class CAccount
     }
 };
 
+static CCriticalSection cs_wallets;
+class CWallets
+{
+private:
+    static std::vector<CWallet *> wallets;
+public:
+    static CWallet* defaultWallet()
+    {
+        LOCK(cs_wallets);
+        if (!wallets.size())
+            return NULL;
+        return wallets[0];
+    }
+    static void addWallet(CWallet *newWallet)
+    {
+        LOCK(cs_wallets);
+        wallets.push_back(newWallet);
+    }
+    static void flushAllWallets(bool shutdown)
+    {
+        LOCK(cs_wallets);
+        for(CWallet *pWallet : wallets)
+            pWallet->Flush(shutdown);
```

Comment:
I decided to hold the lock because it would be possible that during iteration a addWallet (or later removeWallet) happens. But right,.. it results in holding the lock for probably a long time.

I first though doing a copy of the wallets vector would be good, but the CWallet instance where the copy point to might be gone.

Maybe we keep this for now because the flushAllWallets happens only on shutdown.
Or what would be a better solution?